--- vllm/v1/sample/rejection_sampler.py.bak-sampler-0819	2026-08-19 16:59:51.199322427 +0800
+++ vllm/v1/sample/rejection_sampler.py	2026-08-19 16:59:51.205604989 +0800
@@ -471,7 +471,9 @@
             return output_token_ids
 
     # Compute probability distribution from target logits.
-    target_probs = target_logits.softmax(dim=-1, dtype=torch.float32)
+    from vllm.v1.sample.ops.row_softmax import softmax_fp32  # syv patch
+
+    target_probs = softmax_fp32(target_logits)
     assert target_probs.is_contiguous()
 
     # Sample recovered tokens for each position.
@@ -564,7 +566,7 @@
 
     # NOTE(woosuk): `apply_top_k_top_p` uses sorting to calculate the mask,
     # which is slow for large vocab sizes. This may cause performance issues.
-    return apply_top_k_top_p(logits, top_k, top_p)
+    return apply_top_k_top_p(logits, top_k, top_p, k_max=sampling_metadata.top_k_max)
 
 
 def expand_batch_to_tokens(
